pkg/deploy/config.go

```diff
@@ -97,3 +97,26 @@ func mergeConfig(primary, secondary *Configuration) (*Configuration, error) {
 
 	return primary, nil
 }
+
+// CheckRequiredFields validates configuration whether it provides required fields.
+// Config is invalid if required fields are not provided.
+func CheckRequiredFields(config Configuration) error {
```

Comment:
I have moved the check to be part of config structure. I understand the reason, however, I do believe in this case it could have been left as it was. In case of more code, the refactor is a simple one.